cppcheck_junit.py

```diff
@@ -128,14 +140,29 @@ def generate_test_suite(errors: Dict[str, List[CppcheckError]]) -> ElementTree.E
             time=str(1),
         )
         for error in errors:
-            ElementTree.SubElement(
+            error_element = ElementTree.SubElement(
                 test_case,
                 "error",
-                type="",
+                type=error.severity,
                 file=os.path.relpath(error.file) if error.file else "",
-                line=str(error.line),
-                message=f"{error.line}: ({error.severity}) {error.message}",
+                message=f"{error.message}",
             )
+            if len(error.sources) == 0:
+                error_element.text = error.verbose
+            elif len(error.sources) == 1 and error.sources[0][3] == None:
+                source = error.sources[0]
+                file = os.path.relpath(source[0]) if source[0] else ""
+                line = source[1]
+                column = source[2]
+                error_element.text = f"{file}:{line}:{column}: {error.verbose}"
+            else:
+                error_element.text = error.verbose
+                for source in error.sources:
+                    file = os.path.relpath(source[0]) if source[0] else ""
+                    line = source[1]
+                    column = source[2]
+                    info = source[3]
+                    error_element.text += f"\n{file}:{line}:{column}: {info}"
```

Comment:
3 cases adds the rest of the info in the `text` part of the `error` element.
- In the simplest case, just put the `verbose` content.
- If you've a specific `location`, but no extra `info` add `file:line:column: verbose`
- Otherwise, Add `verbose` first, then `file:line:column: info` from all `location`s